components/loginform.js

```diff
@@ -3,14 +3,14 @@ import TwoItemSwitcher from './twoItemSwitcher'
 import '../static/styles/main.scss'
 
 class LoginForm extends Component {
-  constructor () {
-    super()
+  constructor (props) {
+    super(props)
     this.state = {
       name: '',
       user: '',
       pass: '',
       register: false,
-      loginType: 'donor'
+      loginType: props.type === true ? 'teacher' : 'donor'
```

Comment:
If the prop is `type` then the value should be either teacher or donor, not a bool. I recommend you change this `loginType` to match `props.type` directly and change the prop to pass the query string directly.